main.rb

```diff
@@ -0,0 +1,115 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each
+    x = 0
+    while x < length
+      yield(self[x])
+      x += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    result = []
+    my_each { |x| result << x if yield(x) }
+    result
+  end
+
+  def my_all?(*)
+    unless block_given?
+      my_each { |i| return false if i.nil? || i == false }
+      return true
+    end
+    count = 0
+    my_each { |i| count += 1 if yield(i) }
+    count == length
+  end
```

Comment:
## my_all?

- [ ] returns false if the block returns false or nil::range

```Ruby
false_block = proc { |num| num > HIGHEST_VALUE }
range.my_all?(&false_block)
```

_got NameError:
        undefined local variable or method `length' for 5..50:Range_

- [ ] when a class is passed as an argument returns false if any of the collection is not a member of such class

```Ruby
array.my_all?(Integer)
```
_expected false
             got true_


- [ ] when a Regex is passed as an argument returns false if any of the collection does not match the Regex

```Ruby
words.my_all?(/o/)
```

_expected false
             got true_

- [ ] when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern

```Ruby
array.my_all?(3)
```

_expected false
             got true_